### PR TITLE
Expand the Tux ECB image example

### PR DESCRIPTION
**Pull request by nmav**
While cryptographers familiar with Linux know the image and can imagine the original and how it looks in ECB, many who start their path may not fully imagine that process. This commit updates the images to show the original (from wikipedia - converted to png), the ECB-encrypted, and one encrypted with CTR (mentioned as GCM to simplify the story, as CTR is not yet covered).

The penguin image is under CC0 1.0 Universal Public Domain Dedication. The rest are contributed under this project's license.

### CODE REVIEW

**Comment by nmav** [comment]
For simplicity these could also be auto-generated on a system with graphicsmagick and openssl. Let me know if you prefer that path.

PS. That's a course of very impressive quality. Thank you.

File: secure_software_development_fundamentals.md [hunk 1]
```diff
@@ -4499,7 +4499,18 @@ Many symmetric key algorithms, including AES, are what is called *block algorith
 
 The ECB mode is basically a debug or test mode for testing cryptographic algorithms. In ECB mode, the same block of data will produce the same encryption result. This is disastrous for an encryption algorithm, because it reveals far too much about the data that is supposed to be encrypted. A great illustration of this is the so-called “ECB Penguin” image; this image is encrypted using an ECB mode. Encrypted images should appear as random noise, but because ECB mode is used, in the ECB Penguin the image of Tux the Penguin is clearly visible.
```
Comment (david-a-wheeler) [inline secure_software_development_fundamentals.md:4500]: Presuming we do this, the paragraph will need to explain the GCM penguin too. E.g., replace the last sentence with something like:

> Note that in the ECB Penguin below, the image of Tux the Penguin is clearly (and disastrously) visible. An encrypted image *should* appear as random noise, as illustrated by the GCM Penguin below (an encrypted image that uses Galois/Counter mode (GCM)).

File: secure_software_development_fundamentals.md [hunk 2]
```diff
@@ -4499,7 +4499,18 @@ Many symmetric key algorithms, including AES, are what is called *block algorith
...
+    <td>GCM encrypted</td>
+  </tr>
+  <tr>
+    <td><img src="images/penguin.png" width="512" alt="The original Linux Penguin image"></td>
+    <td><img src="images/ecb_penguin.png" width="512" alt="The ECB Penguin: A dark ECB-encrypted image that clearly shows the Linux mascot, Tux the Penguin"></td>
+    <td><img src="images/ctr_penguin.png" width="512" alt="The GCM Penguin: An encrypted image that uses Galois/Counter mode (GCM)"></td>
+  </tr>
+</table>
 
 The ECB Penguin, by Filippo Valsorda, retrieved from [filippo.io](https://blog.filippo.io/the-ecb-penguin/). Licensed under [CC BY-SA 4.0 International](https://creativecommons.org/licenses/by/4.0/legalcode). This image was inspired by the original lower-resolution ECB Penguin image by Wikipedia User: Lunkwill. Source “The ECB Penguin” (2013-11-10). Based on the Tux the penguin official Linux mascot created by Larry Ewing in 1996
```
Comment (david-a-wheeler) [inline secure_software_development_fundamentals.md:4515]: This needs a rework. How about:

> The ECB Penguin and GCM Penguin wer generated by Nikos Mavrogiannopoulos and released under [CC BY-SA 4.0 International](https://creativecommons.org/licenses/by/4.0/legalcode). They were inspired by Filippo Valsorda and the original lower-resolution ECB Penguin image by Wikipedia User: Lunkwill. Source “The ECB Penguin” (2013-11-10). The original image of the Tux the Penguin (official Linux mascot) was created by Larry Ewing in 1996

File: secure_software_development_fundamentals.md (hunk 1 above)
Comment (nmav) [reply secure_software_development_fundamentals.md]: Updated. I've avoided the double parenthesis in the latter sentence if that's ok.

File: secure_software_development_fundamentals.md (hunk 2 above)
Comment (nmav) [reply secure_software_development_fundamentals.md]: Updated. I've included links to Filippo's ecb tux and wikipedia as well.

**Pull request merged**